weld/subsystem/src/main/java/org/jboss/as/weld/WeldCapabilityImpl.java

```diff
@@ -67,11 +67,15 @@ public ServiceBuilder<?> addBeanManagerService(final DeploymentUnit unit, final
     }
 
     public boolean isPartOfWeldDeployment(final DeploymentUnit unit) {
-        return WeldDeploymentMarker.isPartOfWeldDeployment(unit);
+        return forceWeldDeployment || WeldDeploymentMarker.isPartOfWeldDeployment(unit);
     }
 
     public boolean isWeldDeployment(final DeploymentUnit unit) {
-        return WeldDeploymentMarker.isWeldDeployment(unit);
+        return forceWeldDeployment || WeldDeploymentMarker.isWeldDeployment(unit);
+    }
+
+    public void markAsWeldDeployment() {
+        this.forceWeldDeployment = true;
```

Comment:
This should call WeldDeploymentMarker.mark

Reply:
I guess this way can work too, but it seems cleaner to just use the existing WeldDeploymentMarker whose purpose is to track this state.